xhyve/xhyve.go

```diff
@@ -329,7 +330,16 @@ func (d *Driver) GetState() (state.State, error) {
 		return state.Stopped, nil
 	}
 
-	return state.Running, nil
+	psproc, err := ps.FindProcess(int(pid))
+	if err != nil {
+		return state.Error, err
+	}
+	// process name is truncated to 'docker-machine-d'
+	if strings.Contains(psproc.Executable(), "docker-machine") {
```

Comment:
I understand what you mean. failing states first (bailout on not meeting a condition, then based on states of the function) and success state at the bottom. ;-) Will do...